### DISCUSSION TITLE
Inquiry about Editor Functionality

### DISCUSSION THREAD

**Discussion by IDKNWHORU** [toeverything/blocksuite#5838] [Q&A] [answered]
Hello,

I hope this message finds you well. I have a few questions regarding the usage of the editor:

1. Saving Images:
I am interested in learning how to save images. Currently, I am testing the data storage and retrieval functionality using the `job.pageToSnapshot` feature. However, I encounter script errors when trying to load images after saving them. I would appreciate any guidance on resolving this issue.

2. Custom Cloud or Hosting for Image Storage:
When saving images in the editor, I would like to store them at a custom cloud address or on a website that I host. Could you provide information on which API I should use for achieving this?

3. Disabling `new Page` or `import Page` Commands:
I've noticed that when using the new Page or import Page commands, certain features are not available. I would like to display a message when users attempt to utilize these features. Specifically, I aim to either hide them entirely or show a message when someone tries to use the new Page command.

I appreciate your assistance in providing insights or solutions to these queries.

Thank you.

## Reference
- [Image Issue Youtube Channel](https://youtu.be/YEpk7lvCFxM)
- [My Codesandbox Container](https://codesandbox.io/p/github/IDKNWHORU/blocksuite/master?file=%2Fapp%2FEditor.jsx&embed=1&showConsole=true&layout=%257B%2522sidebarPanel%2522%253A%2522GIT%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clqj9rplp0034356amn6jn7yn%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clqj9rplp0031356aukjk0i0q%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clqj9rplp0033356ayu3iimuw%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clqj9rplp0032356a1gnkaxki%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B57.79435078093768%252C42.20564921906232%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clqj9rplp0031356aukjk0i0q%2522%253A%257B%2522id%2522%253A%2522clqj9rplp0031356aukjk0i0q%2522%252C%2522activeTabId%2522%253A%2522clqjacyoc0002356avf5dufcg%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clqj9lxu90001356afl6htpd6%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252F.gitignore%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522type%2522%253A%2522DIFF%2522%252C%2522filepath%2522%253A%2522%252Fapp%252FEditor.jsx%2522%252C%2522base%2522%253A%2522HEAD%2522%252C%2522id%2522%253A%2522clqj9y3xh00os356as7l3c086%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522clqjacyoc0002356avf5dufcg%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A14%252C%2522startColumn%2522%253A41%252C%2522endLineNumber%2522%253A14%252C%2522endColumn%2522%253A41%257D%255D%252C%2522filepath%2522%253A%2522%252Fapp%252FEditor.jsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%257D%252C%2522clqj9rplp0032356a1gnkaxki%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A3000%252C%2522id%2522%253A%2522clqj9ow1x011l356a3kwyy6wf%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clqj9rplp0032356a1gnkaxki%2522%252C%2522activeTabId%2522%253A%2522clqj9ow1x011l356a3kwyy6wf%2522%257D%252C%2522clqj9rplp0033356ayu3iimuw%2522%253A%257B%2522id%2522%253A%2522clqj9rplp0033356ayu3iimuw%2522%252C%2522activeTabId%2522%253A%2522clqjae4ta0093356awfq4vp6x%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522id%2522%253A%2522clqjae4ta0093356awfq4vp6x%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A10.82186394022807%257D)
- [Github Repository](https://github.com/IDKNWHORU/blocksuite)

**Comment by IDKNWHORU** [comment]
Dear everyone,

I sincerely appreciate your assistance. I have successfully addressed the initial two inquiries, namely saving images and exploring options for custom cloud or hosting solutions for image storage.

Now, I kindly seek your guidance on the final question regarding the disabling of new page or import page commands.

Thank you for your continued support.

**Reply by Flrande** [reply, marked as answer]
The methods to override the `linked-page` option and the `slash-menu` option are similar; you can refer to these two links.

https://github.com/toeverything/blocksuite/discussions/5568#discussioncomment-7774850

https://github.com/toeverything/blocksuite/blob/bc7f0abac7cdf6427631d4f95dfac9121b9e3e24/packages/blocks/src/page-block/widgets/linked-page/index.ts#L97

You may need to rewrite a new `getMenu` method for the `linked-page`.

https://github.com/toeverything/blocksuite/blob/bc7f0abac7cdf6427631d4f95dfac9121b9e3e24/packages/blocks/src/page-block/widgets/linked-page/index.ts#L94

https://github.com/toeverything/blocksuite/blob/bc7f0abac7cdf6427631d4f95dfac9121b9e3e24/packages/blocks/src/page-block/widgets/linked-page/config.ts#L69-L146

**Reply by IDKNWHORU** [reply]
Dear @Flrande,

I would like to express my sincere gratitude for providing a great solution. Your guidance on redefining the `slash-menu` was instrumental, and I achieved the desired outcome.

Thank you once again for your valuable assistance.

![image](https://github.com/toeverything/blocksuite/assets/49608580/58026aab-9d38-45bb-9018-d46a9d289cd3)
